### PR TITLE
Observation table sorting/filtering

### PR DESCRIPTION
**Pull request by mcoughlin**
This table adds basic observation table sorting / filtering.

### CODE REVIEW

File: skyportal/handlers/api/observation.py [hunk 1]
```diff
@@ -443,6 +455,78 @@ def get_observations(
...
+                if sort_order == "asc"
+                else [Observation.validity_window_start.desc()]
+            )
+        elif sort_by == "validity_window_end":
+            order_by = (
+                [Observation.validity_window_end]
+                if sort_order == "asc"
+                else [Observation.validity_window_end.desc()]
+            )
+
```
Comment (guynir42) [inline skyportal/handlers/api/observation.py:525]: I would put an `else` with raise value error if it doesn't match any of these strings. 
If not, you can skip the `is not None` condition at the beginning as `None` is not equal to any of these. 
Best is to put the `if order by is None` at the first `if` and the `elif` all the strings and leave the last `else` for a wrong string.

File: skyportal/handlers/api/observation.py [hunk 2]
```diff
@@ -443,6 +455,78 @@ def get_observations(
...
+    if order_by is None:
+        order_by = [Observation.instrument_id.desc()]
+    obs_query = obs_query.order_by(*order_by)
```
Comment (guynir42) [inline skyportal/handlers/api/observation.py:528]: Would this work without the `*` if you also don't put the `order_by` in a list?

**Review by guynir42 — APPROVED** [review]
Minor suggestions...

File: skyportal/handlers/api/observation.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py]: Yes nice thinking.

File: skyportal/handlers/api/observation.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:529]: Good idea.

**Pull request merged**